src/test/nodesessionmetadata_test.cpp

```diff
@@ -210,3 +209,116 @@ TEST_F(NodeSessionMetadataTest, GetSubsessionSizeShouldThrowWhenNonExistingSubse
     auto subsessionMeta = meta.generateSubsessions("request", 5)[0];
     EXPECT_THROW(subsessionMeta.getSubsessionSize("nonExisting"), std::logic_error);
 }
+
+TEST_F(NodeSessionMetadataTest, GetShardIdNoSubsession) {
+    NodeSessionMetadata meta;
+    EXPECT_EQ(meta.getShardId(), 1);
+}
+TEST_F(NodeSessionMetadataTest, GetShardId1SubsessionLevel) {
+    NodeSessionMetadata metaStart;
+    uint subsessionSize = 13;
+    const std::string subsessionName = "subsession";
+    auto subsessions = metaStart.generateSubsessions(subsessionName, subsessionSize);
+    for (size_t i = 0; i < subsessions.size(); ++i) {
+        EXPECT_EQ(subsessions[i].getShardId(), 1);
+    }
+}
+
+TEST_F(NodeSessionMetadataTest, GetShardId1SubsessionLevelCollapsing) {
+    NodeSessionMetadata metaStart;
+    uint subsessionSize = 13;
+    const std::string subsessionName = "subsession";
+    auto subsessions = metaStart.generateSubsessions(subsessionName, subsessionSize);
+    for (size_t i = 0; i < subsessions.size(); ++i) {
+        EXPECT_EQ(subsessions[i].getShardId({subsessionName}), i);
+    }
+}
+
+TEST_F(NodeSessionMetadataTest, GetShardId2SubsessionLevels) {
+    NodeSessionMetadata metaStart;
+    uint subsessionSize1st = 13;
+    uint subsessionSize2nd = 9;
+    const std::string subsessionName1st = "subsession";
+    const std::string subsessionName2nd = "subsession2";
+    auto subsessionsLevel1 = metaStart.generateSubsessions(subsessionName1st, subsessionSize1st);
+    auto subsessionsLevel2 = subsessionsLevel1[4].generateSubsessions(subsessionName2nd, subsessionSize2nd);
+    for (size_t i = 0; i < subsessionsLevel2.size(); ++i) {
+        EXPECT_EQ(subsessionsLevel2[i].getShardId(), 1);
+    }
+}
+
+TEST_F(NodeSessionMetadataTest, GetShardId2SubsessionLevelsCollapse1) {
+    NodeSessionMetadata metaStart;
+    uint subsessionSize1st = 13;
+    uint subsessionSize2nd = 9;
+    const std::string subsessionName1st = "subsession";
+    const std::string subsessionName2nd = "subsession2";
+    auto subsessionsLevel1 = metaStart.generateSubsessions(subsessionName1st, subsessionSize1st);
+    auto subsessionsLevel2 = subsessionsLevel1[4].generateSubsessions(subsessionName2nd, subsessionSize2nd);
```

Comment:
Assert that 4 exists?

Reply:
The same in following tests

Reply:
I think that this would just make test more difficult to understand (additional 2 LoC per generateSubsession - tmp variable + ASSERT_EQ). The size of generated subsessions should be checked by tests above & I added those checks now.